param/parameterized.py

```diff
@@ -1488,21 +1500,32 @@ def trigger(self_, *param_names):
         """
         Trigger watchers for the given set of parameter names. Watchers
         will be triggered whether or not the parameter values have
-        actually changed.
+        actually changed. As a special case, the value will actually be
+        changed for a Parameter of type Event, setting it to True so
+        that it is clear which Event parameter has been triggered.
         """
+        trigger_params = [p for p in self_.self_or_cls.param
+                          if hasattr(self_.self_or_cls.param[p], '_autotrigger_value')]
+        triggers = {p:self_.self_or_cls.param[p]._autotrigger_value
+                    for p in trigger_params if p in param_names}
```

Comment:
Same here:

```python
triggers = {p: self_.self_or_cls.param[p]._autotrigger_value
                    for p in self_.self_or_cls.param if p in param_names and
                    hasattr(self_.self_or_cls.param[p], '_autotrigger_value')}
```

Although this one is more debatable since that dictionary comprehension is getting a bit long.

Reply:
Also can you remind me of the issue if you don't do this and do `set_param` for all of them at once? I'm a little worried about bypassing all this:

```python
        self_.self_or_cls.param._events  = []
        self_.self_or_cls.param._watchers = []
        self_.self_or_cls.param._TRIGGER = True
```

Reply:
I need to remind myself exactly, but I believe I had to split out the trigger parameters from the other ones in other to avoid infinite recursion issues. I'll check whether it works without all these changes just to be sure.

Reply:
Thanks for the suggestion, it looks like the original trigger method now works fine (the issues I was trying to work around must have been due to something else I was trying that is no longer needed). Reverted the trigger method implementation in cfb3518 as I don't see any issues with the original version.

As for the suggested dictionary comprehension, I do think it is a little too long to be worth changing.

Reply:
Looking at the tests, it does seem my changes were necessary after all: the `assert self.e is True` fails in `test_watch_event_trigger_method` without them. Now I just need to trace through the code to remind myself why this is happening!